lib/matplotlib/colors.py

```diff
@@ -1947,3 +1950,38 @@ def from_levels_and_colors(levels, colors, extend='neither'):
 
     norm = BoundaryNorm(levels, ncolors=n_data_colors)
     return cmap, norm
+
+
+def shade_color(color, percent):
+    """A color helper utility to either darken or lighten given color.
+
+    This color utility function allows the user to easily darken or
+    lighten a color for plotting purposes.  This function first
+    converts the given color to RGB using ColorConverter and then to
+    HSL.  The saturation is modified according to the given percentage
```

Comment:
Genuine remark about what may be a typo: it is the lightness ("l" in "hls") that is modified by this function, not the saturation ("s" in "hls"), isn't it?